docs/developing/integrate/wallets/integrating_walletconnect.mdx

```diff
@@ -63,84 +65,81 @@ For this guide, we are configuring a React App.
 
 ```
-yarn add @web3modal/ethereum @web3modal/react wagmi ethers@^5
+yarn add @wagmi/core @web3modal/wagmi viem vite
 ```
 
 </TabItem>
 </Tabs>
 
+3.2 Inside `package.json` file add the following code ( `vite` is a dependency that is starting instant server ):
 
-3.2 Set `projectId={your_project_ID}` in your `.env` file.
+```json
+"scripts": {
+    "dev": "vite"
+}
+```
 
-3.3 Import the following packages in your `app.jsx` file:
+3.3 Create `.env.local` file and place `VITE_PROJECT_ID="XYZ"` ( place your project ID key from [WalletConnect Cloud](https://cloud.walletconnect.com/) ).
 
-```javascript
-import { EthereumClient, w3mConnectors, w3mProvider } from '@web3modal/ethereum'
-import { configureChains, createClient, WagmiConfig } from 'wagmi'
-import { neonDevnet } from '@wagmi/chains'
-
-const chains = [neonDevnet]
-const projectId = "<import_from_env>"
-
-const { provider } = configureChains(chains, [w3mProvider({ projectId })])
-const wagmiClient = createClient({
-    autoConnect: true,
-    connectors: w3mConnectors({ projectId, version: 1, chains }),
-    provider
-})
-```
+3.4 Create `index.html` file and place the following code:
 
-3.4 Pass your ethereumClient and `projectId` to the Web3Modal component:
+```html
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <meta charset="UTF-8" />
+        <title>Neon WalletConnect Example</title>
+    </head>
+    <body>
+        <w3m-button />
+        <script type="module" src="main.js"></script>
+    </body>
+</html>
+```
 
-<Tabs>
-<TabItem value="React/Next" label="app.jsx" default>
+3.5 Create `main.js` file and place the following code:
 
 ```javascript
-import { Web3Modal } from '@web3modal/react'
-
-const web3Modal = new Web3Modal({ projectId }, ethereumClient); // for using in code
-const ethereumClient = new EthereumClient(wagmiClient, chains)
-
-function App() {
-  return <>
-    <WagmiConfig client={wagmiClient}>
-      <HomePage />
-    </WagmiConfig>
-    <Web3Modal projectId={projectId} ethereumClient={ethereumClient} />
-  </>
-}
-```
+import { createWeb3Modal, defaultWagmiConfig } from '@web3modal/wagmi';
+import { neonMainnet, neonDevnet } from '@wagmi/core/chains';
 
-</TabItem>
-<TabItem value="HTML" label="HTML">
+const projectId = import.meta.env.VITE_PROJECT_ID;
+const metadata = {
+    name: 'Neon Example',
+    description: 'Neon Web3Modal Example',
+    url: 'https://neonevm.org',
+    icons: ['https://neonevm.org/favicons/android-chrome-512x512.png']
+};
 
-```html
-<body>
-  <div id="app">
-    <w3m-core-button></w3m-core-button>
-  </div>
-  <script type="module" src="main.js"></script>
-</body>
- ```
+const chains = [neonMainnet, neonDevnet];
+const wagmiConfig = defaultWagmiConfig({chains, projectId, metadata});
 
-</TabItem>
-</Tabs>
+const modal = createWeb3Modal({wagmiConfig, projectId, chains});
+```
 
+3.6 Run command `npm run dev` to start your Web application and you should see the Web3Modal button loaded in your Web app page.
```

Comment:
@mnedelchev-vn previously you used tabs to support yarn command -- pls extend same support here